src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -907,6 +911,33 @@ def _infer_schema_for_file_format(
 
         return new_schema, schema_to_cast, read_file_transformations, None
 
+    def _infer_schema_from_user_input(self, user_schema: StructType, format: str):
+        if format != "JSON":
+            raise ValueError(
+                f"Currently only support user schema for JSON or XML format, got {format} instead"
```

Comment:
the code only handles JSON, why XML mentioned in the error message?

Reply:
we were planning to handle XML as well and then find it impossible, will remove later